src/main/java/graphql/nadel/dsl/ObjectTypeDefinitionWithTransformation.java

```diff
@@ -0,0 +1,129 @@
+package graphql.nadel.dsl;
+
+import graphql.language.Comment;
+import graphql.language.Description;
+import graphql.language.Directive;
+import graphql.language.FieldDefinition;
+import graphql.language.NodeBuilder;
+import graphql.language.ObjectTypeDefinition;
+import graphql.language.SourceLocation;
+import graphql.language.Type;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ObjectTypeDefinitionWithTransformation extends ObjectTypeDefinition {
+
+    private final TypeTransformation typeTransformation;
+
+    protected ObjectTypeDefinitionWithTransformation(String name,
+                                                     List<Type> implementz,
+                                                     List<Directive> directives,
+                                                     List<FieldDefinition> fieldDefinitions,
+                                                     Description description,
+                                                     SourceLocation sourceLocation,
+                                                     List<Comment> comments,
+                                                     TypeTransformation typeTransformation) {
+        super(name, implementz, directives, fieldDefinitions, description, sourceLocation, comments);
+        this.typeTransformation = typeTransformation;
+    }
+
+    public TypeTransformation getTypeTransformation() {
+        return typeTransformation;
+    }
+
+    public static ObjectTypeDefinitionWithTransformation.Builder newObjectTypeDefinitionWithTransformation(ObjectTypeDefinition copyFrom) {
+        return new ObjectTypeDefinitionWithTransformation.Builder(copyFrom);
+    }
+
+    public static final class Builder implements NodeBuilder {
+        private SourceLocation sourceLocation;
+        private List<Comment> comments = new ArrayList<>();
+        private String name;
+        private Description description;
+        private List<Type> implementz = new ArrayList<>();
```

Comment:
what is "implementz",  typo?

Reply:
it is copied from graphql-java: you can't use implements in Java :-)